utils/screeningUtils.js

```diff
@@ -0,0 +1,66 @@
+import { API_BASE } from "../routes/api.js";
+
+
+export async function fetchScreenings() {
+    try {
+        const response = await fetch(`${API_BASE}/screenings?populate=movie`);
+        const screenings = await response.json();
+        return screenings;
+    } catch (error) {
+        console.error('Error fetching screenings:', error);
+        throw error;
+    }
+}
+
+export async function mapScreenings() {
+    const allScreenings = await fetchScreenings();
+
+    const updatedScreenings = allScreenings.data.map(properties => ({
```

Comment:
This is a very nice function, but I have small criticism that can be good to learn from.

The term "map" is well established in programming to mean a type of action where one array is converted to another array. To me as a reader of this code, the name `mapScreenings()` sounds like a function that will take one array of screenings and map it to another array.

You might think at this point, "but that's exactly what this function does!", and you would be correct. But there is one more thing that this function does, which is to _load screenings_.

The name "map screenings" does not reveal that it does something so big, and so significant as loading data from some external source. And because of the use of established terminology, it even implies the opposite, that no big side effects take place.

So I would change this function to take the loaded screenings as a parameter instead.